### PR TITLE
Feat: Migrate the FE to use the new Search GraphQL Queries 

### PR DESCRIPTION
**Pull request by aalises**
Related with #591. Get rid of the old query in favour of the new search queries using the new schema.

### CODE REVIEW

File: apps/core/scenes/results/Results.js [hunk 1]
```diff
@@ -84,6 +72,59 @@ export default class Results extends React.Component<Props> {
...
+            ...ResultsList_data
+          }
+        }
+      `;
+      break;
+  }
+  return query;
+};
+
+const parseSearchParametersByType = (
```
Comment (FilipMessa) [inline apps/core/scenes/results/Results.js]: Can you add this function inside the Component? The reason is consistency.

Comment (aalises) [reply apps/core/scenes/results/Results.js]: Sure! Done

File: apps/graphql/src/apps/itinerary/Itinerary.js [hunk 2]
```diff
@@ -19,20 +19,12 @@ type SearchLocation = {|
...
-  +passengers?: Passengers,
-|};
-
 export type ItinerariesOneWaySearchParameters = {|
   +order?: Order,
   +sort?: Sort,
   +itinerary: {|
     +origin: SearchLocation,
-    +destination?: SearchLocation,
+    +destination: SearchLocation,
```
Comment (FilipMessa) [inline apps/graphql/src/apps/itinerary/Itinerary.js]: The +destination: SearchLocation, is now required?
And for `ItinerariesOneWaySearchParameter` is not? I am confused.

Comment (aalises) [reply apps/graphql/src/apps/itinerary/Itinerary.js]: They are both not required, changed it back. I had changed it to circumvent some Flow issue.

**Pull request merged**